src/gtk/glcanvas.cpp

```diff
@@ -350,4 +354,15 @@ void wxGLCanvas::GTKInitImplicitContext()
 
 #endif // WXWIN_COMPATIBILITY_2_8
 
+void wxGLCanvas::EnsureAvailable()
+{
+#ifdef GDK_WINDOWING_X11
+    if ( !GDK_IS_X11_DISPLAY(gdk_display_get_default()) )
+#endif
+    {
+        wxString msg(wxT("wxGLCanvas is only supported on X11 currently.  You may be able to\nwork around this by setting environment variable GDK_BACKEND=x11 before starting\nyour program."));
+        wxLogFatalError(msg);
```

Comment:
This is, of course, an improvement compared to simply crashing, but could we just return false from here and from `Create()` above and just return to the application after logging a (non-fatal) error? This would seem to be a bit more developer-friendly than just killing the program...

Reply:
In that case, the constructor would just ignore the return value of `Create()` and it would appear as if the construction worked.  Then, I suspect the application would just crash at some later point when it tried to use the `wxGLCanvas.`

Reply:
This is not pretty, to put it mildly, but the canonical way to check for window creation success with wx API is to check for `win->GetHandle() != 0` after creating it. So if this still works here, it's fine.

Even if not, I think it's still better to at least allow using `Create()` without terminating the application...